oauthenticator/auth0.py

```diff
@@ -42,35 +43,64 @@ class Auth0OAuthenticator(OAuthenticator):
 
     login_service = "Auth0"
 
-    auth0_subdomain = Unicode(config=True)
+    auth0_subdomain = Unicode(config=True, allow_none=True)
+    auth0_custom_domain = Unicode(config=True, allow_none=True)
```

Comment:
This PR currently doesn't have an explanation or description to help me with little experience with specifically Auth0. Due to that, reviewing the PR becomes hard.

I see this configuration doesn't include a `help` keyword, but it really should. Also the `auth0_subdomain` should but didn't even though it wasn't added in this PR.

Could you add `help` sections to these configuration options?

Reply:
Looking at the linked issue, I now understand a bit more.

I agree with @GeorgianaElena's comment about deprecating subdomain here: https://github.com/jupyterhub/oauthenticator/issues/207#issuecomment-1076402417.